service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/AssetModuleServiceBean.java

```diff
@@ -75,220 +65,113 @@ public AssetModuleServiceBean(AssetProducerBean assetProducerBean, ActivityConsu
+                    assetQuery.setFlagState(vesselIdentifierStrings);
                     break;
                 case NAME:
-                    assetQuery.setName(vesselIdentifier);
+                    assetQuery.setName(vesselIdentifierStrings);
                     break;
                 case ICCAT:
-                    assetQuery.setIccat(vesselIdentifier);
+                    assetQuery.setIccat(vesselIdentifierStrings);
                     break;
                 case UVI:
-                    assetQuery.setUvi(vesselIdentifier);
+                    assetQuery.setUvi(vesselIdentifierStrings);
                     break;
             }
-        }
-
-        List<AssetDTO> assetList = assetClient.getAssetList(assetQuery);
-
-        return assetList
-                .stream()
-                .map(AssetDTO::getId)
-                .map(UUID::toString)
-                .collect(Collectors.toList());
+        });
+        return assetQuery;
     }
 
     @Override
-    public List<String> getAssetGuids(String vesselSearchStr, String vesselGroupSearch) throws ServiceException {
-        return new ArrayList<>();
-        /*
-        List<String> guidsFromVesselSearchStr = null;
-        List<String> guidsFromVesselGroup     = null;
-        String request;
-
-        // Get the list of guids from assets if vesselSearchStr is provided
-        if (StringUtils.isNotEmpty(vesselSearchStr)) {
-            try {
-                request = AssetModuleRequestMapper.createAssetListModuleRequest(createAssetListQuery(vesselSearchStr));
-            } catch (AssetModelMapperException e) {
-                log.error("Error while trying to map the request for assets Module.", e);
-                throw new ServiceException(e.getMessage(), e.getCause());
-            }
-            guidsFromVesselSearchStr = getGuidsFromAssets(request);
+    public List<String> getAssetGuids(String vesselIdsSearchStr, String vesselGroupGuidSearchStr) throws ServiceException {
+        List<String> resultingGuids = new ArrayList<>();
+
+        // Get the list of guids from assets if vesselIdsSearchStr is provided
+        if (StringUtils.isNotEmpty(vesselIdsSearchStr)) {
+            Map<VesselTypeAssetQueryEnum, List<String>> identifierTypeToValuesMap = createMultipleIdTypesQueryMap(vesselIdsSearchStr);
+            AssetQuery assetQuery = deriveAssetQuery(identifierTypeToValuesMap);
+            boolean dynamic = false;
+            List<AssetDTO> assetList = assetClient.getAssetList(assetQuery, dynamic);
+            resultingGuids.addAll(extractGuidsFromAssets(assetList));
         }
 
         // Get the list of guids from assets if vesselGroupSearchName is provided
-        // If the list of guids is not empty then we have to provide this on the query also
-        if(StringUtils.isNotEmpty(vesselGroupSearch)){
+        if (StringUtils.isNotEmpty(vesselGroupGuidSearchStr)) {
+            UUID vesselGroupGuidSearch = null;
```

Comment:
Maybe just put all three lines in the try{} instead?

Reply:
That's an idea. It would be nicer to the eye, but I do like when you don't have to guess/investigate what can throw the exception.